lib/facter/agent_status_check.rb

```diff
@@ -62,4 +62,25 @@
     x509_cert = OpenSSL::X509::CRL.new(File.read(hostcrl))
     { AS004: (x509_cert.next_update - Time.now) > 7_776_000 }
   end
+  chunk(:AS005) do
+    # Did the last Puppet run have any errors or warnings?
+    last_run_report = Puppet.settings[:lastrunreport]
+    next unless File.exist?(last_run_report)
+    no_errors_or_warnings = true
+    begin
+      last_run_report_file = YAML.load_file(last_run_report)
+      logs = last_run_report_file.logs
+      logs.each do |log|
+        if log.level.to_s == 'warning' || log.level.to_s == 'err'
+          no_errors_or_warnings = false
+          break
+        end
+      end
+    rescue Facter::Core::Execution::ExecutionFailure => e
+      Facter.warn("agent_status_check.AS005 failed to get last run report: #{e.message}")
+      Facter.debug(e.backtrace)
+      { AS005: false }
+    end
+    { AS005: no_errors_or_warnings }
```

Comment:
This line could be replaced with:
```
{ AS005: ! last_run_report_file.logs.any? { |l| [:warning, :err].include?(l.level) } }
```

Then we could remove the `no_errors_or_warnings` variable, the `logs` variable, and the each loop starting on line 73.

Reply:
@Sharpie do you have a moment to take a look at this PR and see whether this fits your suggestion to check for errors or warnings in run reports?